cashapppay/src/main/java/com/adyen/checkout/cashapppay/internal/ui/DefaultCashAppPayDelegate.kt

```diff
@@ -0,0 +1,316 @@
+        } else {
+            CashAppPayFactory.create(componentParams.requireClientId())
+        }.apply {
+            registerForStateUpdates(this@DefaultCashAppPayDelegate)
+        }
+    }
+
+    private fun sendAnalyticsEvent(coroutineScope: CoroutineScope) {
+        Logger.v(TAG, "sendAnalyticsEvent")
+        coroutineScope.launch {
+            analyticsRepository.sendAnalyticsEvent()
+        }
+    }
+
+    override fun observe(
+        lifecycleOwner: LifecycleOwner,
+        coroutineScope: CoroutineScope,
+        callback: (PaymentComponentEvent<CashAppPayComponentState>) -> Unit
+    ) {
+        observerRepository.addObservers(
+            stateFlow = componentStateFlow,
+            exceptionFlow = exceptionChannel.receiveAsFlow(),
+            submitFlow = submitFlow,
+            lifecycleOwner = lifecycleOwner,
+            coroutineScope = coroutineScope,
+            callback = callback
+        )
+    }
+
+    override fun removeObserver() {
+        observerRepository.removeObservers()
+    }
+
+    override fun updateInputData(update: CashAppPayInputData.() -> Unit) {
+        inputData.update()
+        onInputDataChanged()
+    }
+
+    private fun onInputDataChanged() {
+        outputData = createOutputData()
+        updateComponentState(outputData)
+    }
+
+    private fun createOutputData(): CashAppPayOutputData {
+        return CashAppPayOutputData(
+            isStorePaymentSelected = inputData.isStorePaymentSelected,
+            authorizationData = inputData.authorizationData,
+        )
+    }
+
+    @VisibleForTesting
+    internal fun updateComponentState(outputData: CashAppPayOutputData) {
+        Logger.v(TAG, "updateComponentState")
+        val componentState = createComponentState(outputData)
+        _componentStateFlow.tryEmit(componentState)
+    }
+
+    private fun createComponentState(
+        outputData: CashAppPayOutputData = this.outputData
+    ): CashAppPayComponentState {
+        val oneTimeData = outputData.authorizationData?.oneTimeData
+        val onFileData = outputData.authorizationData?.onFileData
+
+        val cashAppPayPaymentMethod = CashAppPayPaymentMethod(
+            type = paymentMethod.type,
+            grantId = oneTimeData?.grantId,
+            customerId = onFileData?.customerId,
+            onFileGrantId = onFileData?.grantId,
+            cashtag = onFileData?.cashTag,
+        )
+
+        val paymentComponentData = PaymentComponentData(
+            paymentMethod = cashAppPayPaymentMethod,
+            order = order,
+            amount = componentParams.amount.takeUnless { it.isEmpty },
+            storePaymentMethod = onFileData != null,
+        )
+
+        return CashAppPayComponentState(
+            data = paymentComponentData,
+            isInputValid = outputData.isValid,
+            isReady = true
+        )
+    }
+
+    override fun onSubmit() {
+        initiatePayment()
```

Comment:
Can we add a check `if (isConfirmationRequired())` to make sure this is only triggered when the switch is visible?